lib.php

```diff
@@ -432,8 +432,9 @@ public function is_plugin_active($cm) {
     public function is_plugin_configured() {
         $turnitinapiurl = get_config('plagiarism_turnitinsim', 'turnitinapiurl');
         $turnitinapikey = get_config('plagiarism_turnitinsim', 'turnitinapikey');
+        $turnitinroutingurl = get_config('plagiarism_turnitinsim', 'turnitinroutingurl');
 
-        return (empty($turnitinapiurl) || empty($turnitinapikey)) ? false : true;
+        return (!empty($turnitinapikey) && (!empty($turnitinapiurl) || !empty($turnitinroutingurl))) ? true : false;
```

Comment:
Wouldn't we always have both turnitinapiurl and turnitinroutingurl if one of them is defined?

Reply:
Hi Mat,
This wouldn't be the case on a fresh install, configuring the plugin for the first time. If they configure with the API URL it will have to make a call to get the routing URL.

Reply:
Sorry I asked my question wrong. If turnitinroutingurl is defined, turnitinapiurl should always be defined, right? We don't remove it when we find the routing URL. (Because we want to show the API URL still.) Or is it that we allow for the user to enter the routing URL when doing the setup?